kotlinx-coroutines-test/common/src/TestBuilders.kt

```diff
@@ -228,7 +229,26 @@ internal suspend fun <T: AbstractCoroutine<Unit>> runTestCoroutine(
                     // we expect these and will instead throw a more informative exception just below.
                     emptyList()
                 }.throwAll()
-                throw UncompletedCoroutinesError("The test coroutine was not completed after waiting for $dispatchTimeoutMs ms")
+                var completing: Boolean
+                val completionCause = try {
+                    coroutine.tryGetCompletionCause().also { completing = true }
+                } catch (e: Throwable) {
+                    completing = false
```

Comment:
instead of `try-catch`, it should be enough to check `coroutine.isCancelled`.
The same goes for `completing`